ecommerce/main.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+)
+var productName string
+
+type Product struct {
+	name       string
+	price      float64
+	stock      bool
+	stockCount int
+}
+
+func (p *Product) Buy() {
+	if p.stock && p.stockCount > 0 {
+		p.stockCount--
+		fmt.Printf("Purchased %s. New stock count: %d\n", p.name, p.stockCount)
+		if p.stockCount <=0{
+			p.stock = false;
+		}
+	} else {
+		p.stock=false;
+		fmt.Printf("Product %s is out of stock.\n", p.name)
+	}
+}
+
+func searchProduct (products []Product ,name string){
+	for _,product := range products{
+		if product.name == name{
+			fmt.Printf("Your Product %+v",product)
```

Comment:
Burada slice içindeki eşleşen product'u bulduktan sonra artık slice üzerinde gezmesine gerek kalmadığı için döngüyü kırabilirsin (break veya return ile)